### PR TITLE
565 mylastworkflows

### PR DESCRIPTION
**Pull request by ricardovdheijden**
#565 

Adds a summary card on the start page listing the 5 latest workflows the the logged in user

### CODE REVIEW

**Review by wouter1975 — APPROVED** [review]

File: packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx [hunk 1]
```diff
@@ -111,7 +120,27 @@ export const WfoStartPage = () => {
...
+        headerStatus: SummaryCardStatus.Success,
+        listTitle: t('myWorkflows.listTitle'),
+        listItems:
+            myWorkflowsSummaryResponse?.processes.map((workflow) => ({
+                title: workflow.workflowName,
+                value: formatDate(workflow?.startedAt),
+                url: `${PATH_WORKFLOWS}/${workflow.processId}`,
+            })) ?? [],
+        button: username
+            ? {
```
Comment (wouter1975) [inline packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx:138]: I am in doubt whether we should hide the summary card if oath is disabled. I guess most deployments will have oath enabled, but for the example-orchestrator-ui, this card will useless...

File: packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx [hunk 2]
```diff
@@ -111,7 +120,27 @@ export const WfoStartPage = () => {
...
+        isLoading: activeWorkflowsSummaryIsFetching,
+    };
+
+    const myWorkflowsSummaryCard: SummaryCard = {
+        headerTitle: t('myWorkflows.headerTitle'),
+        headerValue: myWorkflowsSummaryResponse?.pageInfo.totalItems ?? 0,
+        headerStatus: SummaryCardStatus.Success,
+        listTitle: t('myWorkflows.listTitle'),
+        listItems:
+            myWorkflowsSummaryResponse?.processes.map((workflow) => ({
```
Comment (tjeerddie) [inline packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx:132]: this is the same mapping as `activeWorkflowsSummaryResponse?.processes.map`

File: packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx [hunk 3]
```diff
@@ -217,30 +247,59 @@ const outOfSyncSubscriptionsListSummaryQueryVariables: GraphqlQueryVariables<Sub
...
-            value: 'false',
-        },
-        {
-            field: 'lastStatus',
-            value: 'created-running-suspended-waiting-failed-resumed-inconsistent_data-api_unavailable-awaiting_callback',
+const getMyWorkflowListSummaryQueryVariables = (
+    username: string,
+): GraphqlQueryVariables<Process> => {
+    return {
+        first: 5,
```
Comment (tjeerddie) [inline packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx:254]: move some fields in a default object, to use in each `...QueryVariables` and move isTask filter to its own variable so you only add the Todo once.
```
return {
    ...defaultQueryVars,
    filterBy: [filterByNotTask, ...]
}
```

**Review by tjeerddie — APPROVED** [review]

File: packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx (hunk 2 above)
Comment (mrijk) [inline packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx:132]: This code looks a lot like the code on line 114. Maybe extract the common functionality in a separate component?

File: packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx (hunk 1 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx:138]: Done

File: packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx (hunk 2 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx:132]: Done

Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/pages/startPage/WfoStartPage.tsx:132]: Done

File: packages/orchestrator-ui-components/src/messages/en-GB.json [hunk 4]
```diff
@@ -372,6 +372,11 @@
         }
     },
     "startPage": {
+        "myWorkflows": {
+            "buttonText": "Show my completed workflows",
+            "headerTitle": "My complete workflows",
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/messages/en-GB.json:377]:
```suggestion
            "headerTitle": "My completed workflows",
```

File: packages/orchestrator-ui-components/src/utils/optionalArray.spec.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { optionalArrayMapper, toOptionalArrayEntry } from './optionalArray';
+
+describe('toOptionalArrayEntry', () => {
+    const testInput = { testField: 'testValue' };
+
+    it('returns an array with the data if ', () => {
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/utils/optionalArray.spec.ts:6]: if.... ?

**Review by DutchBen — APPROVED** [review]

**Pull request merged**